### ISSUE TITLE
`cast`ing between ints and float has different behaviour on JS backend

### ISSUE THREAD

**Issue by konsumlamm** [nim-lang/Nim#22326]
### Description

On the C backend, `cast`ing between ints and floats just reinterprets the underlying bits. On the JS backend however, it rounds the numbers, since (non-64-bit) ints are represented as JS `Number`s, which are floats.

As far as I can tell, we could achieve the same behaviour as on the C backend for the JS backend by using arrays.

Is this desirable? Or should we just document the different behaviour?

### Nim Version

Nim Compiler Version 1.9.5 [Linux: amd64]
Compiled at 2023-07-22
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: b02c1dd6ca96548b47d978f96278c67bf59e9d9e
active boot switches: -d:release

**Comment by juancarlospaco** [comment]
`--jsbigint64:on` ?.

**Comment by konsumlamm** [comment]
Yes, it's on by default, but this also happens with other integer types.
